onnxscript/rewriter/_pattern_ir.py

```diff
@@ -76,20 +76,28 @@ def __str__(self) -> str:
 class AttrPattern(Pattern[ir.Attr]):
     """Base class for an attribute pattern. Matches any attribute value by default."""
 
-    def __init__(self, name: str | None):
+    def __init__(self, name: str | None, *, can_match_none: bool = False):
         self._name = name
+        self._can_match_none = can_match_none
 
     @property
     def name(self) -> str | None:
         return self._name
 
+    @property
+    def can_match_none(self) -> bool:
+        """Indicates whether this pattern can match a None attribute."""
+        return self._can_match_none
+
     def matches(self, attr: ir.Attr) -> bool:
         return True
 
     def __str__(self) -> str:
         return self._name if self._name is not None else "anonymous:" + str(id(self))
 
 
+AttrVar = AttrPattern
```

Comment:
Could you comment on this alias? Why is it needed?

Reply:
`AttrVar` is a better name for a pattern-variable meant to be bound against an attribute-value.  `AttrPattern` serves as the base-class for other attribute-patterns as well, such as attribute-constants. The names also mirror those used for ValuePattern and Var. (In the long run, we could make AttrVar a derived class of AttrPattern, but right now there is no need, so an alias works fine.)

Reply:
I am thinking maybe a derived class is better for documentation, because right now AttrVar will not have the right docstring I assume

Reply:
Done.